### ISSUE TITLE
iotune: LeakSanitizer: detected memory leaks

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#7596]
*Installation details*
Scylla version (or git commit hash): scylla-unified-package-4.4.dev.0.20201110.78c598e08e.tar.gz
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS8 instance on GCE
GCE Machine type: e2-medium (2 vCPUs, 4 GB memory)

### Description

In the manual test of nonroot offline install, a memory leak was detected in scylla_io_setup.

```
Do you want IOTune to study your disks IO profile and adapt Scylla to it? (*WARNING* Saying NO here means the node will not boot in production mode unless you configure the I/O Subsystem manually!)
Yes - let iotune study my disk(s). Note that this action will take a few minutes. No - skip this step.
[YES/no]YES
ERROR:root:This is not a recommended Google Cloud instance setup for auto tuning, running manual iotune.
tuning /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/sda2
tuning /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda
tuning: /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/queue/nomerges 2
warning: unable to tune /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/queue/nomerges to 2

tuning /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/sda2
tuning /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/sda2
tuning /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/sda2
tuning /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/sda2
WARNING: debug mode. Not for benchmarking or production
WARN  2020-11-12 08:27:30,908 [shard 0] seastar - Unable to set SCHED_FIFO scheduling policy for timer thread; latency impact possible. Try adding CAP_SYS_NICE
==43891==WARNING: ASan doesn't fully support makecontext/swapcontext functions and may produce false positives in some cases!
INFO  2020-11-12 08:27:31,619 [shard 0] iotune - /home/scylla-test/scylladb/saved_caches passed sanity checks
WARN  2020-11-12 08:27:31,623 [shard 0] iotune - nomerges for /sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/queue/nomerges set to 0. It is recommend to set it to 2 before evaluation so that merges are disabled. Results can be skewed otherwise.
Starting Evaluation. This may take a while...
Measuring sequential write bandwidth: 375 MB/s
Measuring sequential read bandwidth: 121 MB/s
Measuring random write IOPS: 321 IOPS
Measuring random read IOPS: 151 IOPS
Writing result to /home/scylla-test/scylladb/etc/scylla.d/io_properties.yaml
Writing result to /home/scylla-test/scylladb/etc/scylla.d/io.conf

=================================================================
==43891==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 4096 byte(s) in 1 object(s) allocated from:
    #0 0x5c0b77  (/home/scylla-test/scylladb/libexec/iotune+0x5c0b77)
    #1 0xa2bb2a  (/home/scylla-test/scylladb/libexec/iotune+0xa2bb2a)
    #2 0x6042e7  (/home/scylla-test/scylladb/libexec/iotune+0x6042e7)
    #3 0x602987  (/home/scylla-test/scylladb/libexec/iotune+0x602987)
    #4 0x6028d7  (/home/scylla-test/scylladb/libexec/iotune+0x6028d7)
    #5 0x60281b  (/home/scylla-test/scylladb/libexec/iotune+0x60281b)
    #6 0x602764  (/home/scylla-test/scylladb/libexec/iotune+0x602764)
    #7 0x6024df  (/home/scylla-test/scylladb/libexec/iotune+0x6024df)
    #8 0x6021bc  (/home/scylla-test/scylladb/libexec/iotune+0x6021bc)
    #9 0x601f54  (/home/scylla-test/scylladb/libexec/iotune+0x601f54)
    #10 0xa32cd0  (/home/scylla-test/scylladb/libexec/iotune+0xa32cd0)
    #11 0x1187846  (/home/scylla-test/scylladb/libexec/iotune+0x1187846)
    #12 0x11870bd  (/home/scylla-test/scylladb/libexec/iotune+0x11870bd)
    #13 0x7fe350bbc28f  (/home/scylla-test/scylladb/libreloc/libc.so.6+0x5328f)

SUMMARY: AddressSanitizer: 4096 byte(s) leaked in 1 allocation(s).
ERROR:root:Command '['/home/scylla-test/scylladb/bin/iotune', '--format', 'envfile', '--options-file', '/home/scylla-test/scylladb/etc/scylla.d/io.conf', '--properties-file', '/home/scylla-test/scylladb/etc/scylla.d/io_properties.yaml', '--evaluation-directory', '/home/scylla-test/scylladb/data', '--evaluation-directory', '/home/scylla-test/scylladb/commitlog', '--evaluation-directory', '/home/scylla-test/scylladb/hints', '--evaluation-directory', '/home/scylla-test/scylladb/view_hints', '--evaluation-directory', '/home/scylla-test/scylladb/saved_caches']' returned non-zero exit status 1.
ERROR:root:['/home/scylla-test/scylladb/data', '/home/scylla-test/scylladb/commitlog', '/home/scylla-test/scylladb/hints', '/home/scylla-test/scylladb/view_hints', '/home/scylla-test/scylladb/saved_caches'] did not pass validation tests, it may not be on XFS and/or has limited disk space.
This is a non-supported setup, and performance is expected to be very bad.
```

### Logs

1. install log: [nonroot-offline-install-log.txt](https://github.com/scylladb/scylla/files/5529181/nonroot-offline-install-log.txt)
2. setup log: 
[scylla_setup-log.txt](https://github.com/scylladb/scylla/files/5529230/scylla_setup-log.txt)
3. systemd log: 
[systemd-log.txt](https://github.com/scylladb/scylla/files/5529231/systemd-log.txt)

/CC @avikivity @roydahan

**Comment by avikivity** [comment]
It is pretty minor since iotune is not a server, so if it leaks 4k, nobody cares. 

Nevertheless, @xemul please place it at the bottom of your queue, in the small cupboard at the left, in the middle drawer.

**Comment by slivne** [comment]
But why are we having the santizer running - is this a debug build ?

**Comment by amoskong** [comment]
> But why are we having the santizer running - is this a debug build ?

It's not a debug build.

- http://downloads.scylladb.com.s3.amazonaws.com/relocatable/unstable/master/2020-11-10T00%3A01%3A15Z/scylla-unified-package-4.4.dev.0.20201110.78c598e08e.tar.gz

**Comment by avikivity** [comment]
It is a debug build, otherwise you wouldn't have anything from leaksanitizer.

Maybe there is a mixup in package upload and the debug package found its way there.

**Comment by slivne** [comment]
the debug was fixed #7653
